aiida/orm/nodes/data/structure.py

```diff
@@ -503,14 +503,11 @@ def get_formula(symbol_list, mode='hill', separator=''):
     if mode in ['hill', 'hill_compact']:
         symbol_set = set(symbol_list)
         first_symbols = []
-        if 'C' in symbol_set:
-            # remove C (and H if present) from list and put them at the
-            # beginning
-            symbol_set.remove('C')
-            first_symbols.append('C')
-            if 'H' in symbol_set:
-                symbol_set.remove('H')
-                first_symbols.append('H')
+        for special_symbol in ['C', 'H']:
+            # remove C and H if present from list and put them at the beginning
+            if special_symbol in symbol_set:
+                symbol_set.remove(special_symbol)
+                first_symbols.append(special_symbol)
         ordered_symbol_set = first_symbols + list(sorted(symbol_set))
```

Comment:
Assuming that the interpretation of the Hill notation is correct, I think this code could be simplified:
```suggestion
        ordered_symbol_set = list(sorted(set(symbol_list), key=lambda s: {'C': '0', 'H': '1'}.get(s, s)))
```

Reply:
It isn't, unfortunately.
Just pushed the update (reverting to previous implementation).

Reply:
Could still be massively simplified?
```python
if 'C' in symbol_list:
    ordered_symbol_set = list(sorted(set(symbol_list), key=lambda s: {'C': '0', 'H': '1'}.get(s, s))
else:
    ordered_symbol_set = list(sorted(set(symbol_list)))
```
Or am I missing something?

Reply:
This is a really nice condensation, however!